lib/Viroverse/Controller/NumericAssayProtocol.pm

```diff
@@ -0,0 +1,47 @@
+use strict;
+use warnings;
+use utf8;
+use 5.018;
+
+package Viroverse::Controller::NumericAssayProtocol;
+use Moose;
+use Try::Tiny;
+use Catalyst::ResponseHelpers;
+use namespace::autoclean;
+
+BEGIN { extends 'Viroverse::Controller' }
+
+sub base : Chained('/') PathPart('numeric-assay-protocol') CaptureArgs(0) {
+    my ($self, $c) = @_;
+    return Forbidden($c)
+        unless $c->stash->{scientist}->is_admin;
+}
+
+sub create : POST Chained('base') PathPart('create') Args(0) {
+    my ($self, $c) = @_;
+    return ClientError($c, "Must supply a protocol name")
+        unless defined $c->req->params->{name};
```

Comment:
For whatever reason this test isn't sufficient to detect that no protocol is being specified -- when I left the dropdown unselected I got the error:

```
Couldn't add assay result: DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR: null value in column "numeric_assay_protocol_id" violates not-null constraint DETAIL: Failing row contains (3, null, 106903, 244, null, null, null, 2019-08-07, 2019-08-07 23:58:00.949422+00). [for Statement "INSERT INTO viroserve.numeric_assay_result ( date_completed, note, numeric_assay_protocol_id, sample_id, scientist_id, uri, value) VALUES ( ?, ?, ?, ?, ?, ?, ? ) RETURNING numeric_assay_result_id" with ParamValues: 1='2019-08-07', 2=undef, 3=undef, 4='106903', 5='244', 6=undef, 7=undef]
```